sycl/test/extensions/properties/new_properties_open_question.cpp

```diff
@@ -0,0 +1,185 @@
+// RUN: %clangxx -fsycl -fsyntax-only %s
+
+#include <sycl/ext/oneapi/properties/properties.hpp>
+
+using namespace sycl::ext::oneapi::experimental::new_properties;
+
+template <typename property_key_t> constexpr auto generate_property_key_name() {
+#if defined(__clang__) || defined(__GNUC__)
+  return __PRETTY_FUNCTION__;
+#elif defined(_MSC_VER)
+  return __FUNCSIG__;
+#else
+#error "Unsupported compiler"
+#endif
+}
+
+template <typename property_t, typename property_key_t = property_t>
+struct named_property_base
+    : public detail::property_base<property_t, property_key_t> {
+  static constexpr std::string_view property_name =
+      generate_property_key_name<property_key_t>();
+};
+
+struct prop_key {};
+
+template <typename PropList> auto foo(PropList props = {}) {
+  static_assert(props.template has_property<prop_key>());
+  static_assert(props.template get_property<prop_key>().value == 42);
+}
+
+namespace approach_one {
+// Don't provide `inline constexpr` property variables at all, they don't
+// provide enough benefits now that `property_value` thing was eliminated.
+
+template <int N> struct prop : named_property_base<prop<N>, prop_key> {
+  static constexpr int value = N;
+};
+struct other_prop : named_property_base<other_prop> {};
+
+void bar() {
+  foo(properties{prop<42>{}});
+  foo(properties{prop<42>{}, other_prop{}});
+}
+} // namespace approach_one
+
+namespace approach_two {
+// Keep providing `inline constexpr` "shortcuts".
+template <int N>
+struct prop_property : named_property_base<prop_property<N>, prop_key> {
+  static constexpr int value = N;
+};
+struct other_prop_property : named_property_base<other_prop_property> {};
+
+template <int N> inline constexpr prop_property<N> prop{};
+inline constexpr other_prop_property other_prop{};
+
+void bar() {
+  // Still need `properties` here.
+  foo(properties{prop<42>});
+  foo(properties{prop<42>, other_prop});
+}
+} // namespace approach_two
+namespace approach_three {
+// Use operator+ to create `properties` property list from individual
+// properties.
+template <int N>
+struct prop_property : named_property_base<prop_property<N>, prop_key> {
+  static constexpr int value = N;
+};
+struct other_prop_property : named_property_base<other_prop_property> {};
+
+template <int N> inline constexpr prop_property<N> prop{};
+inline constexpr other_prop_property other_prop{};
+
+void bar() {
+  // Unary `+` has a bit of hacky feeling...
+  foo(+prop<42>);
+  // More than one property in a property list looks very natural.
+  // Alternatively, that can be `operator|` but it has no unary version.
+  foo(prop<42> + other_prop);
```

Comment:
We want to provide a simplified user interface like this, but it's unclear what's the best way to implement this. This file sketches few possible approaches starting from here.